drivers/i3c/i3c_stm32.c

```diff
@@ -0,0 +1,2171 @@
+	if (ret != 0) {
+		LOG_ERR("Error sending IBI ENEC for 0x%02x (%d)", target->dynamic_addr, ret);
+	}
+
+	/* Set I3C bus devices configuration */
+	LL_I3C_ConfigDeviceCapabilities(i3c, (idx + 1), target->dynamic_addr,
+					LL_I3C_IBI_CAPABILITY, LL_I3C_IBI_DATA_ENABLE,
+					LL_I3C_CR_NO_CAPABILITY);
+
+	return ret;
+}
+
+int i3c_stm32_ibi_disable(const struct device *dev, struct i3c_device_desc *target)
+{
+	int ret = 0;
+	uint8_t idx;
+	I3C_TypeDef *i3c;
+	struct i3c_ccc_events i3c_events;
+	struct i3c_stm32_data *data = dev->data;
+	const struct i3c_stm32_config *config = dev->config;
+
+	i3c = config->i3c;
+	if (!i3c_device_is_ibi_capable(target)) {
+		return -EINVAL;
+	}
+
+	for (idx = 0; idx < ARRAY_SIZE(data->ibi.addr); idx++) {
+		if (target->dynamic_addr == data->ibi.addr[idx]) {
+			break;
+		}
+	}
+
+	if (idx == ARRAY_SIZE(data->ibi.addr)) {
+		LOG_ERR("%s: target is not in list of registered addresses", __func__);
+		return -ENODEV;
+	}
+
+	data->ibi.addr[idx] = 0U;
+	data->ibi.num_addr -= 1U;
+
+	if (data->ibi.num_addr == 0U) {
+		(void)pm_device_runtime_put(dev);
+	}
+
+	/* Tell target to disable IBI */
+	i3c_events.events = I3C_CCC_EVT_INTR;
+	ret = i3c_ccc_do_events_set(target, false, &i3c_events);
+	if (ret != 0) {
+		LOG_ERR("Error sending IBI DISEC for 0x%02x (%d)", target->dynamic_addr, ret);
+	}
+
+	if ((target != NULL) && i3c_device_is_ibi_capable(target)) {
+		/* Set I3C bus devices configuration */
+		LL_I3C_ConfigDeviceCapabilities(i3c, (idx + 1), target->dynamic_addr,
+						LL_I3C_IBI_NO_CAPABILITY,
+						i3c_ibi_has_payload(target) ? LL_I3C_IBI_DATA_ENABLE : LL_I3C_IBI_DATA_DISABLE,
+						LL_I3C_CR_NO_CAPABILITY);
+	}
+
+	return ret;
+}
+
+#endif /* CONFIG_I3C_USE_IBI */
+
+#ifdef CONFIG_I3C_STM32_DMA
+static void i3c_stm32_tx_rx_msg_config(const struct device *dma_dev, void *user_data,
+				       uint32_t channel, int status)
+{
+	const struct device *dev = (const struct device *)user_data;
+
+	if (i3c_stm32_curr_msg_xfer_next(dev) != 0) {
+		/* No more messages to transmit/receive */
+		return;
+	}
+
+	uint8_t *buf = NULL;
+	size_t *offset = 0;
+	uint32_t len = 0;
+
+	i3c_stm32_curr_msg_xfer_get_buf(dev, &buf, &len, &offset);
+	i3c_stm32_dma_msg_config(dev, (uint32_t)buf, len);
+}
+
+static void i3c_stm32_dma_tx_cb(const struct device *dma_dev, void *user_data, uint32_t channel,
+				int status)
+{
+	i3c_stm32_tx_rx_msg_config(dma_dev, user_data, channel, status);
+}
+
+static void i3c_stm32_dma_rx_cb(const struct device *dma_dev, void *user_data, uint32_t channel,
+				int status)
+{
+	i3c_stm32_tx_rx_msg_config(dma_dev, user_data, channel, status);
+}
+
+static void i3c_stm32_dma_tc_cb(const struct device *dma_dev, void *user_data, uint32_t channel,
+				int status)
+{
+}
+
+static void i3c_stm32_dma_rs_cb(const struct device *dma_dev, void *user_data, uint32_t channel,
+				int status)
+{
+}
+
+#endif
+
+static const struct i3c_driver_api i3c_stm32_driver_api = {
```

Comment:
Device driver APIs need to use the `DEVICE_API` wrapper.

```suggestion
static DEVICE_API(i3c, i3c_stm32_driver_api) = {
```

Reply:
@ExaltZephyr Note that this is a new macro and this may require a rebase.